dashboard/app/controllers/api/v1/users_controller.rb

```diff
@@ -112,7 +112,10 @@ def dismiss_census_banner
 
   # POST /api/v1/users/<user_id>/dismiss_donor_teacher_banner
   def dismiss_donor_teacher_banner
-    @user.donor_teacher_banner_dismissed = true
+    @user.donor_teacher_banner_dismissed = {
+      participate: params[:participate].to_s == "true",
+      source: params[:source]
```

Comment:
I think it'd be slightly more rails-y to use [strong parameters](https://edgeguides.rubyonrails.org/action_controller_overview.html#strong-parameters) here? But you're not doing a mass assignment, so in terms of security I don't think it matters.